rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -108,9 +102,14 @@ def get_agent_status(self,
 
         results_object = self._post_payload(payload)
 
-        agent = results_object.get("data").get("assets")[0].get("agent")
-        quarantine_state = agent.get("quarantineState").get("currentState")
-        agent_status = agent.get("agentStatus")
+        try:
+            agent = results_object.get("data").get("assets")[0].get("agent")
+            quarantine_state = agent.get("quarantineState").get("currentState")
+            agent_status = agent.get("agentStatus")
+        except KeyError:
```

Comment:
```suggestion
        except (KeyError, IndexError):
```
Key for dictionaries, index for lists